### PR TITLE
Redirect debug output to `defined` destination

### PR DESCRIPTION
**Pull request by thesebas**
Allow redirecting debug output to somewhere else than STDOUT - like STDERR or any other stream/file (like Monolog stream wrapper)

### CODE REVIEW

**Comment by michaelklishin** [comment]
This makes a lot of sense. @nubeiro @hairyhum any objections to merging it?

File: PhpAmqpLib/Helper/DebugHelper.php [hunk 1]
```diff
@@ -101,6 +107,6 @@ public function debug_connection_start($version_major, $version_minor, $server_p
      * @param string $s
      */
     protected function print_msg($s) {
-        echo $s . PHP_EOL;
+        fprintf($this->debug_output, $s . PHP_EOL);
```
Comment (hairyhum) [inline PhpAmqpLib/Helper/DebugHelper.php]: `fprintf` expects a format string and arguments to be provided, so the semantics is different from `echo` (in presence of `%` symbol). Should it be a `fwrite` instead?

Comment (thesebas) [inline PhpAmqpLib/Helper/DebugHelper.php]: I agree, `fwrite` is better here.

**Comment by thiagotalma** [comment]
Hi guys!

This change triggered this exception:
`Use of undefined constant STDOUT - assumed 'STDOUT'`

@thesebas @michaelklishin @nubeiro

**Comment by michaelklishin** [comment]
@thiagotalma that's a little light on details. In what environment, with what kind of configuration? Feel free to submit a PR that sets the default to be standard output that'd avoid warnings and such.

**Comment by thiagotalma** [comment]
https://github.com/php-amqplib/php-amqplib/pull/501  ;)

**Pull request merged**